### ISSUE TITLE
Install failed

### ISSUE THREAD

**Issue by chriswaudby** [varioustoxins/NEF-Pipelines#3]
Just tried to install with pipx on nmrbox but got an error with `nef_pipelines.transcoders.nmrstar`:

```
cwaudby@argon:~$ pipx install nef-pipelines
  installed package nef-pipelines 0.1.81, installed using Python 3.8.10
  These apps are now globally available
    - nef
done! ✨ 🌟 ✨
cwaudby@argon:~$ nef
Usage: nef [OPTIONS] COMMAND [ARGS]...

Options:
  --install-completion [bash|zsh|fish|powershell|pwsh]
                                  Install completion for the specified shell.
  --show-completion [bash|zsh|fish|powershell|pwsh]
                                  Show completion for the specified shell, to
                                  copy it or customize the installation.
  --help                          Show this message and exit.

Commands:
  chains     - carry out operations on chains
  csv        - read [rdcs]
  deep       - read deep [peaks]
  echidna    - read echidna data [peaks]
  entry      - carry out operations on the nef file entry
  fasta      - read and write fasta sequences
  fit        - carry out fitting operations [alpha]
  frames     - carry out operations on frames in nef frames
  globals    - add global options to the pipeline [use save as your last...
  header     - add a header to the stream
  help       - help on the nef pipelines tools and their usage
  mars       - read and write mars [shifts and sequences]
  modelfree  - write modelfree [relaxation data]
  nmrpipe    - read nmrpipe [peaks shifts & sequencess]
  nmrstar    - read NMR-STAR [sequences & shifts]
  nmrview    - read and write nmrview [peaks, sequences & shifts]
  pales      - read and write pales/dc [rdcs]
  pdbx       - read pdb [sequences]
  peaks      - carry out operations on nef peaks
  rpf        - write rpf shifts
  save       - save the entries in the stream to a file / files or stdout...
  series     - carry out operations on a data series
  shifts     - carry out operations on shifts [average]
  shiftx2    - read shiftx2 data [shifts]
  shifty     - write shifty [shifts]
  simulate   - simulate data
  sink       - read the current stream and don't write anything
  sparky     - read sparky files [shifts]
  stream     - stream a nef file
  talos      - read and write talos files [shifts & restraints]
  test       - run the test suite
  xcamshift  - write xcamshift for xplor [shifts]
  xeasy      - read xeasy files [flya dialect: sequence]
  xplor      - read xplor [sequences, dihedral & distance restraints]

*************************************************************** error in nef_pipelines.transcoders.nmrstar *****************************************************************

plugin nef_pipelines.transcoders.nmrstar
Traceback (most recent call last):
  File "/home/nmrbox/cwaudby/.local/share/pipx/venvs/nef-pipelines/lib/python3.8/site-packages/nef_pipelines/main.py", line 116, in main
    import_module(module_name)
  File "/usr/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 848, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/nmrbox/cwaudby/.local/share/pipx/venvs/nef-pipelines/lib/python3.8/site-packages/nef_pipelines/transcoders/nmrstar/__init__.py", line 22, in <module>
    import nef_pipelines.transcoders.nmrstar.importers.project  # noqa: F401
  File "/home/nmrbox/cwaudby/.local/share/pipx/venvs/nef-pipelines/lib/python3.8/site-packages/nef_pipelines/transcoders/nmrstar/importers/project.py", line 28, in <module>
    from nef_pipelines.transcoders.nmrstar.importers.shifts import pipe as shift_pipe
  File "/home/nmrbox/cwaudby/.local/share/pipx/venvs/nef-pipelines/lib/python3.8/site-packages/nef_pipelines/transcoders/nmrstar/importers/shifts.py", line 28, in <module>
    from nef_pipelines.lib.translation.chem_comp import ID, ChemCompLinking
  File "/home/nmrbox/cwaudby/.local/share/pipx/venvs/nef-pipelines/lib/python3.8/site-packages/nef_pipelines/lib/translation/chem_comp.py", line 8, in <module>
    from typing import (
ImportError: cannot import name 'Annotated' from 'typing' (/usr/lib/python3.8/typing.py)

****************************************************************************************************************************************************************************

WARNING: the following plugins failed to load:

    nef_pipelines.transcoders.nmrstar

the remaining plugins will work but NEF-Pipelines is working  with reduced capabilities. The modules
that failed to load and the causes of the problem are listed above please report this to the authors
at github: https://github.com/varioustoxins/NEF-Pipelines/issues
```

**Comment by varioustoxins** [comment]
Hi Chris

That’s a bug on my part, for 3.8 to work I need to import annotations from future __future__

If you put

from __future__ import annotations

At the top of the file giving errors (chem_comp.py)  it may sort it

I will add a test to catch this and a fix hopefully later today or tomorrow

Another alternative is to point pipx at a later python

Thank you for the bug report and I hope that helps

regards
Gary




Dr Gary S Thompson NMR Facility Manager

CCPN CoI & Working Group Member
Wellcome Trust Biomolecular NMR Facility
School of Biosciences,  Division of Natural Sciences
University of Kent,  Canterbury,  Kent,  England,  CT2 7NZ

☎:01227 82 7117
✉️: ***@***.***
orchid: orcid.org/0000-0001-9399-7636

On 24 Jul 2024, at 13:07, Chris Waudby ***@***.***> wrote:


CAUTION: This email originated from outside of the organisation. Do not click links or open attachments unless you recognise the sender and know the content is safe.




Just tried to install with pipx on nmrbox but got an error with nef_pipelines.transcoders.nmrstar:

***@***.***:~$ pipx install nef-pipelines
  installed package nef-pipelines 0.1.81, installed using Python 3.8.10
  These apps are now globally available
    - nef
done! ✨ 🌟 ✨
***@***.***:~$ nef
Usage: nef [OPTIONS] COMMAND [ARGS]...

Options:
  --install-completion [bash|zsh|fish|powershell|pwsh]
                                  Install completion for the specified shell.
  --show-completion [bash|zsh|fish|powershell|pwsh]
                                  Show completion for the specified shell, to
                                  copy it or customize the installation.
  --help                          Show this message and exit.

Commands:
  chains     - carry out operations on chains
  csv        - read [rdcs]
  deep       - read deep [peaks]
  echidna    - read echidna data [peaks]
  entry      - carry out operations on the nef file entry
  fasta      - read and write fasta sequences
  fit        - carry out fitting operations [alpha]
  frames     - carry out operations on frames in nef frames
  globals    - add global options to the pipeline [use save as your last...
  header     - add a header to the stream
  help       - help on the nef pipelines tools and their usage
  mars       - read and write mars [shifts and sequences]
  modelfree  - write modelfree [relaxation data]
  nmrpipe    - read nmrpipe [peaks shifts & sequencess]
  nmrstar    - read NMR-STAR [sequences & shifts]
  nmrview    - read and write nmrview [peaks, sequences & shifts]
  pales      - read and write pales/dc [rdcs]
  pdbx       - read pdb [sequences]
  peaks      - carry out operations on nef peaks
  rpf        - write rpf shifts
  save       - save the entries in the stream to a file / files or stdout...
  series     - carry out operations on a data series
  shifts     - carry out operations on shifts [average]
  shiftx2    - read shiftx2 data [shifts]
  shifty     - write shifty [shifts]
  simulate   - simulate data
  sink       - read the current stream and don't write anything
  sparky     - read sparky files [shifts]
  stream     - stream a nef file
  talos      - read and write talos files [shifts & restraints]
  test       - run the test suite
  xcamshift  - write xcamshift for xplor [shifts]
  xeasy      - read xeasy files [flya dialect: sequence]
  xplor      - read xplor [sequences, dihedral & distance restraints]

*************************************************************** error in nef_pipelines.transcoders.nmrstar *****************************************************************

plugin nef_pipelines.transcoders.nmrstar
Traceback (most recent call last):
  File "/home/nmrbox/cwaudby/.local/share/pipx/venvs/nef-pipelines/lib/python3.8/site-packages/nef_pipelines/main.py", line 116, in main
    import_module(module_name)
  File "/usr/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 848, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/nmrbox/cwaudby/.local/share/pipx/venvs/nef-pipelines/lib/python3.8/site-packages/nef_pipelines/transcoders/nmrstar/__init__.py", line 22, in <module>
    import nef_pipelines.transcoders.nmrstar.importers.project  # noqa: F401
  File "/home/nmrbox/cwaudby/.local/share/pipx/venvs/nef-pipelines/lib/python3.8/site-packages/nef_pipelines/transcoders/nmrstar/importers/project.py", line 28, in <module>
    from nef_pipelines.transcoders.nmrstar.importers.shifts import pipe as shift_pipe
  File "/home/nmrbox/cwaudby/.local/share/pipx/venvs/nef-pipelines/lib/python3.8/site-packages/nef_pipelines/transcoders/nmrstar/importers/shifts.py", line 28, in <module>
    from nef_pipelines.lib.translation.chem_comp import ID, ChemCompLinking
  File "/home/nmrbox/cwaudby/.local/share/pipx/venvs/nef-pipelines/lib/python3.8/site-packages/nef_pipelines/lib/translation/chem_comp.py", line 8, in <module>
    from typing import (
ImportError: cannot import name 'Annotated' from 'typing' (/usr/lib/python3.8/typing.py)

****************************************************************************************************************************************************************************

WARNING: the following plugins failed to load:

    nef_pipelines.transcoders.nmrstar

the remaining plugins will work but NEF-Pipelines is working  with reduced capabilities. The modules
that failed to load and the causes of the problem are listed above please report this to the authors
at github: https://github.com/varioustoxins/NEF-Pipelines/issues

**Comment by varioustoxins** [comment]
fixed by commit  [79604b4](https://github.com/varioustoxins/NEF-Pipelines/commit/1291a371a7798bbc4479a0ba54eb3e1076af743b) which both fixes the problem and adds tests to ensure the main program starts and all plugins load during testing on GitHub pipelines. Fix published in version 0.1.82 distribution.
